activation/activation.go

```diff
@@ -654,3 +667,38 @@ func buildNipostChallengeStartDeadline(roundStart time.Time, gracePeriod time.Du
 	jitter := randomDurationInRange(time.Duration(0), gracePeriod*maxNipostChallengeBuildJitter/100.0)
 	return roundStart.Add(jitter).Add(-gracePeriod)
 }
+
+func findFullyValidHighTickAtx(
```

Comment:
I don't think this is necessary. What can go wrong in your opinion without a transaction? Note: a full chain validation can take a considerable amount of time (it doubles every epoch) because until malfeasance proofs sync is fixed we should probably validate the whole chain back to epoch 1.

Reply:
The function is only used for a) the commitment ATX, where time shouldn't matter and for the b) positioning ATX. Especially for the later we want to avoid concurrent calls to this function that might cause double validation of the same ATXs (in the case of multi-smesher) - or we only call this function once per CG and use the same positioning ATX for all identities of the node.

Regarding double effort: it only doubles every epoch if the new ATX has a chain consisting solely of ATXs that go all the way back to the golden ATX without including a single ATX that was already validated. This becomes less and less likely the more epochs back you go; Nodes try to pick the highest positioning ATX they can there will only be so many unique positioning ATXs in an epoch and eventually every chain you find will go through that subset.

Reply:
Let's add to the handler that validating all indices immediately sets ATXs to valid during sync, so that a node doesn't have to validate the whole chain when selecting a positioning ATX during the cyclegap, to avoid running late for the registration of the next poet round.

Since a race is unlikely we can leave not doing the looking in a single TX. 🙂

Reply:
Done